gtk4/src/subclass/font_chooser.rs

```diff
@@ -0,0 +1,375 @@
+        self.parent_set_font_map(font_chooser, font_map)
+    }
+
+    fn font_map(&self, font_chooser: &Self::Type) -> Option<FontMap> {
+        self.parent_font_map(font_chooser)
+    }
+
+    fn font_activated(&self, font_chooser: &Self::Type, font_name: &str) {
+        self.parent_font_activated(font_chooser, font_name)
+    }
+}
+
+pub trait FontChooserImplExt: ObjectSubclass {
+    fn parent_font_family(&self, font_chooser: &Self::Type) -> Option<FontFamily>;
+    fn parent_font_face(&self, font_chooser: &Self::Type) -> Option<FontFace>;
+    fn parent_font_size(&self, font_chooser: &Self::Type) -> i32;
+    fn parent_set_filter_func(&self, font_chooser: &Self::Type, callback: Option<&FilterCallback>);
+    fn parent_set_font_map<P: IsA<FontMap>>(&self, font_chooser: &Self::Type, font_map: Option<&P>);
+    fn parent_font_map(&self, font_chooser: &Self::Type) -> Option<FontMap>;
+    fn parent_font_activated(&self, font_chooser: &Self::Type, font_name: &str);
+}
+
+impl<O: FontChooserImpl> FontChooserImplExt for O {
+    fn parent_font_family(&self, font_chooser: &Self::Type) -> Option<FontFamily> {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface = type_data.as_ref().parent_interface::<FontChooser>()
+                as *const ffi::GtkFontChooserIface;
+            let f = (*parent_iface)
+                .get_font_family
+                .expect("no parent \"get_font_family\" implementation");
+
+            from_glib_none(f(font_chooser
+                .unsafe_cast_ref::<FontChooser>()
+                .to_glib_none()
+                .0))
+        }
+    }
+
+    fn parent_font_face(&self, font_chooser: &Self::Type) -> Option<FontFace> {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface = type_data.as_ref().parent_interface::<FontChooser>()
+                as *const ffi::GtkFontChooserIface;
+
+            let f = (*parent_iface)
+                .get_font_face
+                .expect("no parent \"get_font_face\" implementation");
+
+            from_glib_none(f(font_chooser
+                .unsafe_cast_ref::<FontChooser>()
+                .to_glib_none()
+                .0))
+        }
+    }
+
+    fn parent_font_size(&self, font_chooser: &Self::Type) -> i32 {
+        unsafe {
+            let type_data = Self::type_data();
+            let parent_iface = type_data.as_ref().parent_interface::<FontChooser>()
+                as *const ffi::GtkFontChooserIface;
+
+            if let Some(f) = (*parent_iface).get_font_size {
+                f(font_chooser
+                    .unsafe_cast_ref::<FontChooser>()
+                    .to_glib_none()
+                    .0)
+            } else {
+                // No font size is selected
+                -1
+            }
+        }
+    }
+
+    fn parent_set_filter_func(&self, font_chooser: &Self::Type, callback: Option<&FilterCallback>) {
```

Comment:
And this needs to consume the callback (i.e. also no reference) so that you don't get a double free